kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Config.java

```diff
@@ -270,7 +270,7 @@ private boolean tryKubeConfig(Config config) {
               config.setPassword(currentAuthInfo.getPassword());
 
               config.getErrorMessages().put(401, "Unauthorized! Token may have expired! Please log-in again.");
-              config.getErrorMessages().put(403, "Forbidden! User "+config.getUsername()+ " doesn't have permission.");
+              config.getErrorMessages().put(403, "Forbidden! User "+currentContext.getUser()+ " doesn't have permission.");
```

Comment:
From https://github.com/kubernetes/kubernetes/blob/f4beccf000c706fc66644422062e91391576da8a/pkg/client/unversioned/clientcmd/api/types.go#L109-L110 `currentContext.getUser()` should return the `name of the authInfo for this context`, not the username:

> // AuthInfo is the name of the authInfo for this context
	AuthInfo string `json:"user"`

The username is embedded in the `AuthInfo` object that this references - if there is no matching `AuthInfo` object then there is something wrong with the kube config file. I would prefer to log that fact & keep the username logic the same as it currently is.